pkg/dbs/studydb/db_study-participants.go

```diff
@@ -168,3 +170,51 @@ func (dbService *StudyDBService) DeleteMessagesFromParticipant(instanceID string
 	_, err := dbService.collectionRefStudyParticipant(instanceID, studyKey).UpdateOne(ctx, filter, update)
 	return err
 }
+
+func (dbService *StudyDBService) CreateMessageScheduledForIndex(instanceID string, studyKey string) error {
+	ctx, cancel := dbService.getContext()
+	defer cancel()
+
+	_, err := dbService.collectionRefStudyParticipant(instanceID, studyKey).Indexes().CreateOne(
+		ctx, mongo.IndexModel{
+			Keys: bson.D{
+				{Key: "messages.scheduledFor", Value: 1},
+			},
+		},
+	)
+	return err
+}
+
+func (dbService *StudyDBService) CreateMessageScheduledForIndexForAllStudies(instanceID string) {
+	studies, err := dbService.GetStudiesByStatus(instanceID, "", true)
+	if err != nil {
+		logger.Error.Printf("unexpected error when fetching studies in '%s': %v", instanceID, err)
+		return
+	}
+
+	for _, study := range studies {
+		err = dbService.CreateMessageScheduledForIndex(instanceID, study.Key)
+		if err != nil {
+			logger.Error.Printf("unexpected error when creating survey definition indexes: %v", err)
+		}
+	}
+}
+
+func (dbService *StudyDBService) CheckParticipantsForPendingMessages(instanceID string, studyKey string) (hasMessage bool, err error) {
+	ctx, cancel := dbService.getContext()
+	defer cancel()
+
+	filter := bson.M{"studyStatus": types.PARTICIPANT_STUDY_STATUS_ACTIVE}
```

Comment:
@kschneider84 please check if we need compound index rather than just index for messages.scheduledFor (here query/filter has two attributes, studyStatus and messages.schedulerFor

Reply:
@phev8 added compound index in method